### PR TITLE
Add and process building include attribute

### PR DESCRIPTION
**Pull request by bloodypenguin**
Refs #24

### CODE REVIEW

File: BuildingThemes/Configuration.cs [hunk 1]
```diff
@@ -72,15 +73,20 @@ public class Building
...
             public Building(string name)
             {
                 this.name = name;
             }
 
-            public Building(string name, bool builtIn)
+            public Building(string name, bool isBuiltIn, bool include)
             {
                 this.name = name;
-                this.isBuiltIn = true;
```
Comment (earalov) [inline BuildingThemes/Configuration.cs]: Hope this fix doesn't break any of existing code relying on this behaviour

File: BuildingThemes/Configuration.cs [hunk 2]
```diff
@@ -72,15 +73,20 @@ public class Building
...
+                this.isBuiltIn = isBuiltIn;
```
Comment (bloodypenguin) [inline BuildingThemes/Configuration.cs:87]: Hope this fix doesn't break any of existing code relying on this behavior

Comment (boformer) [inline BuildingThemes/Configuration.cs:87]: This  was actually always set to true?

Comment (bloodypenguin) [inline BuildingThemes/Configuration.cs:87]: yes it was

File: BuildingThemes/Configuration.cs [hunk 3]
```diff
@@ -35,13 +36,13 @@ public class Theme
             [XmlArrayItem(ElementName = "Building")]
             public List<Building> buildings = new List<Building>();
 
-            public void addAll(string[] buildingNames, bool builtIn)
+            public void addAll(string[] buildingNames, string[] removedBuildings, bool builtIn)
```
Comment (boformer) [inline BuildingThemes/Configuration.cs]: You don't need to change this. The foreach statement below will only add buildings that are not already in the list (see containsBuilding method). And the buildings which were disabled by the user are already in the list when the vanilla buildings are added.

Comment (bloodypenguin) [inline BuildingThemes/Configuration.cs]: Right. I'll fix that.

File: BuildingThemes/Configuration.cs [hunk 4]
```diff
@@ -72,15 +73,18 @@ public class Building
             [XmlIgnoreAttribute]
             public bool isBuiltIn = false;
 
+            [XmlAttribute(" include"), DefaultValue(true)]
```
Comment (boformer) [inline BuildingThemes/Configuration.cs]: the space character in this string causes that the config is not loaded correctly.

**Comment by boformer** [comment]
Still needs more testing

**Comment by boformer** [comment]
And we have to remove that debug log call... I forgot that it was there.

**Comment by boformer** [comment]
Works perfectly! Merging...

**Pull request merged**